test/fast_layers.jl

```diff
@@ -38,6 +44,12 @@ fsgrad = Flux.Zygote.gradient((x,p)->sum(fs(x,p)),x,pd)
 @test fdgrad[1] ≈ fsgrad[1]
 @test fdgrad[2] ≈ fsgrad[2] rtol=1e-5
 
+fdcgrad = Flux.Zygote.gradient((x,p)->sum(fdc(x,p)),x,pd)
+@test fdgrad[1] ≈ fdcgrad[1]
+@test fdgrad[2] ≈ fdcgrad[2] rtol=1e-12
+@allocated(Zygote.pullback(fdc, x, pd))
+@test @allocated(Zygote.pullback(fdc, x, pd)) < 1450

```

Comment:
decrease?

Reply:
Now this 1376 is due to the closure(FastDense_adjoint) which we can't deal with right now

Reply:
Yup okay, that's good. Tidy up the rest of the tests and I think it's good to merge.